src/test/server/store-test.js

```diff
@@ -1,9 +1,203 @@
+            });
+        });
+
+        describe('updateUser', () => {
+            it('Should emit change event', async () => {
+                assert.notEqual(UserStore.getUsers()[1], UPDATE_USER);
+                await subEvent(EVENT_TYPE.change, () => UserAction.updateUser(UPDATE_USER));
+
+                assert.deepEqual(UserStore.getUsers()[1], UPDATE_USER);
+            });
+
+            it('Should emit usersFound event with updated foundUsers and searchFields if update in search mode', async () => {
+                assert.notEqual(UserStore.getUsers()[1], UPDATE_USER);
+
+                UserStore._searchedInfo = SEARCH_USER_INFO;
+
+                const [updatedFoundUsers, ...args] = await subEvent(EVENT_TYPE.usersFound, () => UserAction.updateUser(UPDATE_USER));
+
+                assert.strictEqual(updatedFoundUsers.length, 0);
+                assert.deepEqual(UserStore.getUsers()[1], UPDATE_USER);
+            });
+        });
+
+        describe('findUser', () => {
+            it('Should emit usersFound event with empty foundUsers and searchFields if users not found', async () => {
+                const [foundUsers, searchFields] = await subEvent(EVENT_TYPE.usersFound, () => UserAction.findUser(SEARCH_NON_EXIST_USER_INFO));
+
+                assert.deepEqual(UserStore._searchedInfo, SEARCH_NON_EXIST_USER_INFO);
+                assert.equal(foundUsers.length, 0);
+                assert.deepEqual(searchFields, SEARCH_FIELDS);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if user found', async () => {
+                const [foundUsers, searchFields] = await subEvent(EVENT_TYPE.usersFound, () => UserAction.findUser(SEARCH_USER_INFO));
+
+                assert.deepEqual(foundUsers, [UserStore.getUsers()[1]]);
+                assert.deepEqual(searchFields, SEARCH_FIELDS);
+            });
+        });
+
+        describe('stopFindUser', () => {
+            it('Should emit change event and clear searchedUser if stop search user', async () => {
+                await subEvent(EVENT_TYPE.change, () => UserAction.stopFindUser());
+
+                assert.strictEqual(UserStore._searchedInfo, null);
+            });
+        });
+    });
+
+    describe('private methods', () => {
+        describe('_findUserIndexById', () => {
+            it('Should return the index of the user with the same id', () => {
+                assert.strictEqual(UserStore._findUserIndexById(UPDATE_USER.id), 1);
+            });
+        });
+
+        describe('_findUserIndexByName', () => {
+            it('Should return the index of the user with the same name', () => {
+                assert.deepEqual(UserStore._findUserIndexByName(TEST_USER_EXISTING.name), 1);
+            });
+        });
+
+        describe('_defineSearchFields', () => {
+            it('Should return the index of the user with the same name', () => {
+                UserStore._searchedInfo = new UserInfo({
```

Comment:
why do we need to assign a new `UserInfo` object to a class property?
we can use just a const here